File: year2/systems-design-implementation/Lab9/SwimModel/src/main/java/domain/Inscriere.java
package domain;

import utils.Entity;
import utils.Pair;

import java.io.Serializable;

public class Inscriere implements Serializable {

    private Integer id_participant;
    private Integer id_proba;
    private Participant participant;
    private Proba proba;

    public Inscriere(Integer id_participant, Integer id_proba, Participant participant, Proba proba) {
        this.id_participant = id_participant;
        this.id_proba = id_proba;
        this.participant = participant;
        this.proba = proba;
    }

    public Inscriere(Integer id_participant, Integer id_proba) {
        this.id_participant = id_participant;
        this.id_proba = id_proba;
        this.participant = null;
        this.proba = null;
    }

    public Integer getId_participant() {
        return id_participant;
    }

    public void setId_participant(Integer id_participant) {
        this.id_participant = id_participant;
    }

    public Integer getId_proba() {
        return id_proba;
    }

    public void setId_proba(Integer id_proba) {
        this.id_proba = id_proba;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipant(Participant participant) {
        this.participant = participant;
    }

    public Proba getProba() {
        return proba;
    }

    public void setProba(Proba proba) {
        this.proba = proba;
    }

    @Override
    public String toString() {
        return "domain.Inscriere{" +
                "participant=" + participant +
                ", proba=" + proba +
                '}';
    }
}

File: year2/systems-design-implementation/Lab6/SwimModel/Domain/Inscriere.cs
using SwimModel.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SwimModel.Domain
{
    [Serializable]
    public class Inscriere : Entity<Pair<int, int>>
    {
        public Participant participant { get; set; }
        public Proba proba { get; set; }
        public Inscriere(int e1, int e2)
        {
            base.id = new Pair<int, int>(e1, e2);
            participant = null;
            proba = null;
        }
        public Inscriere(int e1, int e2, Participant pa, Proba pr)
        {
            base.id = new Pair<int, int>(e1, e2);
            participant = pa;
            proba = pr;
        }
    }
}
